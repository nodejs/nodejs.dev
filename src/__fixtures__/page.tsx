import {
  PaginationInfo,
  LearnTemplateContext,
  NavigationData,
  PostTemplateData,
  PostTemplateContext,
  NodeReleaseData,
  PageTableOfContents,
  BlogCategory,
  BlogPost,
  ArticleData,
} from '../types';
import mockMDXBodyContent from './mockMDXBodyContent';

export const mockTableOfContents: PageTableOfContents = {
  items: [
    { title: 'mock-title', url: '#mock-title' },
    { title: 'mock-title-2', url: '#mock-title-2' },
  ],
};

export const createPaginationInfo = (): PaginationInfo =>
  ({
    slug: 'test-slug',
    title: 'test-title',
  } as PaginationInfo);

export const createNavigationSectionData = (): NavigationData =>
  ({
    'test-section': [
      {
        slug: 'test-slug-1',
        title: 'test-title-1',
      },
      {
        title: 'test-title-2',
        slug: 'test-slug-2',
      },
    ],
    'test-section2': [
      {
        slug: 'test-slug-3',
        title: 'test-title-3',
      },
      {
        title: 'test-title-4',
        slug: 'test-slug-4',
      },
    ],
  } as NavigationData);

export const createLearnPageData = () => ({
  data: {
    articleCurrentLanguage: {
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
      frontmatter: {
        title: 'test-title',
        description: 'test-description',
      },
      fields: {
        authors: ['test-user1', 'test-user2'],
      },
    } as ArticleData,
  },
});

export const createNodeReleasesData = (): NodeReleaseData[] =>
  [
    {
      endOfLife: '2022-04-30',
      ltsStart: '2019-10-21',
      maintenanceStart: '2020-11-30',
      initialRelease: '2019-04-23',
      codename: 'erbium',
      version: 'v12',
      fullVersion: 'v12.0.0',
      status: 'Maintenance LTS',
      isLts: false,
    },
    {
      endOfLife: '2023-04-30',
      ltsStart: '2020-10-17',
      maintenanceStart: '2021-10-19',
      initialRelease: '2020-04-21',
      codename: 'fermium',
      version: 'v14',
      fullVersion: 'v14.0.0',
      status: 'Active LTS',
      isLts: true,
    },
    {
      endOfLife: '2025-04-30',
      ltsStart: '2022-10-25',
      maintenanceStart: '2023-10-18',
      initialRelease: '2022-04-19',
      codename: 'v18',
      version: 'v18',
      fullVersion: 'v18.0.0',
      status: 'Current',
      isLts: false,
    },
  ] as NodeReleaseData[];

export const createLearnPageContext = (): LearnTemplateContext =>
  ({
    slug: 'test-slug',
    relativePath: 'test-path',
    next: createPaginationInfo(),
    previous: createPaginationInfo(),
    navigationData: createNavigationSectionData(),
  } as LearnTemplateContext);

export const createBlogPageContext = (): PostTemplateContext => ({
  slug: 'test-slug',
  relativePath: 'test-path',
  next: createPaginationInfo(),
  previous: createPaginationInfo(),
  navigationData: createNavigationSectionData(),
  recent: [
    {
      node: {
        frontmatter: {
          title: 'title-mock',
          category: {
            name: 'category-mock',
            slug: 'category-mock-slug',
          },
          blogAuthors: [
            {
              id: 'id-mock',
              name: 'name-mock',
              website: 'url-mock',
            },
          ],
        },
        fields: {
          date: '11/11/2022',
          slug: 'slug-mock',
          readingTime: { text: 'text-mock' },
        },
      },
    },
  ],
});

export const createBlogData = (): {
  posts: { edges: BlogPost[] };
  categories: { edges: BlogCategory[] };
} => ({
  posts: {
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Mock blog title',
            category: {
              name: 'mock-category',
              slug: 'Mock Category Slug',
            },
            blogAuthors: [
              {
                name: 'Mock author name',
                website: 'Mock URL',
              },
            ],
          },
          fields: {
            date: '11/11/2022',
            slug: 'Mock blog slug',
            readingTime: {
              text: '1 min read',
            },
          },
        },
      },
    ],
  },
  categories: {
    edges: [],
  },
});

export const createBlogPageData = (): PostTemplateData => ({
  mdx: {
    body: mockMDXBodyContent,
    excerpt: 'excerpt-mock',
    frontmatter: {
      title: 't,itle-mock',
      blogAuthors: [
        {
          id: 'id-mock',
          name: 'name-mock',
          website: 'url-mock',
        },
      ],
    },
    fields: { slug: 'slug-mock', date: '11/11/2022' },
  },
});

export const createGeneralPageData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['author-mock'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createResourcesData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['MrJithil'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createPrivacyData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['Author'],
      },
      frontmatter: {
        description: 'Description',
        title: 'Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});
