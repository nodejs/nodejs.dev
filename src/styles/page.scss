.page > .article-reader {
  margin: 0 auto;
}
