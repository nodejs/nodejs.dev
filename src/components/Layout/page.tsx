import React from 'react';
import Article from '../Article';
import Layout from '.';

import '../../styles/page.scss';

interface Page {
  editPath?: string;
  html: string;
  tableOfContents: string;
  title: string;
  description: string;
  authors: string[];
  location?: Location;
}

export default function PageLayout(props: Page): JSX.Element {
  const {
    editPath,
    html,
    tableOfContents,
    title,
    description,
    authors,
    location,
  } = props;
  return (
    <Layout title={title} location={location} description={description}>
      <main className="page">
        <Article
          html={html}
          title={title}
          authors={authors}
          editPath={editPath}
          tableOfContents={tableOfContents}
        />
      </main>
    </Layout>
  );
}
