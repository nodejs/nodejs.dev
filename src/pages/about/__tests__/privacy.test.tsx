import React from 'react';
import { render } from '@testing-library/react';
import PrivacyPage from '../privacy';
import { createPrivacyData } from '../../../__fixtures__/page';

const mockData = createPrivacyData();

describe('Privacy Page', () => {
  it('renders correctly', () => {
    const { container } = render(<PrivacyPage data={mockData.data} />);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });
});
