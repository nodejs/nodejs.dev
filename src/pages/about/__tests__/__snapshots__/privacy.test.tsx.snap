// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy Page renders correctly 1`] = `
<main
  class="grid-container"
>
  <nav
    class="sideNav"
  >
    <button
      class="sideNavOpen"
      type="button"
    >
      Menu
    </button>
    <div
      class="sideNavSection"
    >
      <div
        class="sideNavSectionTitle"
      >
        <b>
          Community Pages
        </b>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-8rv7ug-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
      <a
        class="t-body2 navigationItem"
        href="/about/"
        id="link-/about/"
      >
        About
      </a>
      <a
        class="t-body2 navigationItem"
        href="/community/"
        id="link-/community/"
      >
        Community
      </a>
      <a
        class="t-body2 navigationItem"
        href="/download/package-manager/"
        id="link-/download/package-manager/"
      >
        Package Manager
      </a>
      <a
        aria-current="page"
        class="t-body2 navigationItem navigationItemActive"
        href="/about/privacy/"
        id="link-/about/privacy/"
      >
        Privacy Policy
      </a>
      <a
        class="t-body2 navigationItem"
        href="/about/governance/"
        id="link-/about/governance/"
      >
        Project Governance
      </a>
      <a
        class="t-body2 navigationItem"
        href="/about/releases/"
        id="link-/about/releases/"
      >
        Releases
      </a>
      <a
        class="t-body2 navigationItem"
        href="/resources/"
        id="link-/resources/"
      >
        Resources
      </a>
      <a
        class="t-body2 navigationItem"
        href="/about/security/"
        id="link-/about/security/"
      >
        Security Reporting
      </a>
      <a
        class="t-body2 navigationItem"
        href="/about/working-groups/"
        id="link-/about/working-groups/"
      >
        Working Groups
      </a>
    </div>
  </nav>
  <article
    class="article"
  >
    <h1
      class="headline"
    >
      Title
    </h1>
    <details
      class="tableOfContents"
    >
      <summary>
        <strong>
          Table of Contents
        </strong>
      </summary>
      <ul>
        <li>
          <a
            href="#mock-title"
          >
            mock-title
          </a>
        </li>
        <li>
          <a
            href="#mock-title-2"
          >
            mock-title-2
          </a>
        </li>
      </ul>
    </details>
    <div>
      <p>
        Node.js can be installed in different ways. This post highlights the most common and convenient ones.
      </p>
      <p>
        Official packages for all the major platforms are available at 
        <a>
          https://nodejs.dev/download/
        </a>
        .
      </p>
      <p>
        One very convenient way to install Node.js is through a package manager. In this case, every operating system has its own.
      </p>
      <p>
        Other package managers for MacOS, Linux, and Windows are listed in 
        <a>
          https://nodejs.dev/download/package-manager/
        </a>
      </p>
      <p>
        <code
          class="code"
        >
          nvm
        </code>
         is a popular way to run Node.js. It allows you to easily switch the Node.js version, and install new versions to try and easily rollback if something breaks, for example.
      </p>
      <p>
        It is also very useful to test your code with old Node.js versions.
      </p>
      <p>
        See 
        <a>
          https://github.com/nvm-sh/nvm
        </a>
         for more information about this option.
      </p>
      <p>
        In any case, when Node.js is installed you'll have access to the 
        <code
          class="code"
        >
          node
        </code>
         executable program in the command line.
      </p>
    </div>
    <div
      class="authorList"
    >
      Contributors
      <ul>
        <li>
          <a
            class="link"
            href="https://github.com/Author"
            rel="noopener noreferrer"
            style="margin-left: 0px;"
            target="_blank"
            title="Author"
          >
            <img
              alt="Author"
              src="https://github.com/Author.png?size=60"
            />
          </a>
        </li>
      </ul>
    </div>
    <div
      class="edit"
    >
      <a
        href="https://github.com/nodejs/nodejs.dev/edit/main/content/about/privacy.en.md"
      >
        <span>
          Edit this page on GitHub
        </span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pencil "
          data-icon="pencil"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.7 220.3l-11.3 11.3-22.6 22.6-205 205c-6.6 6.6-14.8 11.5-23.8 14.1L30.8 511c-8.4 2.5-17.5 .2-23.7-6.1S-1.5 489.7 1 481.2L38.7 353.1c2.6-9 7.5-17.2 14.1-23.8l205-205 22.6-22.6 11.3-11.3 33.9 33.9 62.1 62.1 33.9 33.9zM96 353.9l-9.3 9.3c-.9 .9-1.6 2.1-2 3.4l-25.3 86 86-25.3c1.3-.4 2.5-1.1 3.4-2l9.3-9.3H112c-8.8 0-16-7.2-16-16V353.9zM453.3 19.3l39.4 39.4c25 25 25 65.5 0 90.5l-14.5 14.5-22.6 22.6-11.3 11.3-33.9-33.9-62.1-62.1L314.3 67.7l11.3-11.3 22.6-22.6 14.5-14.5c25-25 65.5-25 90.5 0z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </article>
</main>
`;
